pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/results/BaseResultsBlock.java

```diff
@@ -155,16 +158,16 @@ public void setNumServerThreads(int numServerThreads) {
     _numServerThreads = numServerThreads;
   }
 
+  public abstract DataSchema getDataSchema(QueryContext queryContext);
+
+  public abstract Collection<Object[]> getRows(QueryContext queryContext)
```

Comment:
(nit) brief javadoc please and also highlight how this is added for optimizing the SerDe aspect